### PR TITLE
Add the missing cmake file for parallel stl

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: cmake/FindParallelSTL.cmake [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        if(ParallelSTL_TBB_HAS_EXECUTION_PAR)
+            set(ParallelSTL_USES_TBB On)
+            set(ParallelSTL_LIBRARIES TBB::tbb)
+            message(STATUS "Using TBB for parallel execution")
+        endif()
+    endif()
+endif()
+
+foreach(VAR ParallelSTL_FOUND ParallelSTL_LIBRARIES ParallelSTL_USES_TBB)
+    string(TOUPPER ${VAR} ParallelSTL_VAR)
```
Comment (umangyadav) [inline cmake/FindParallelSTL.cmake:42]: why is this necessary ?

Comment (pfultz2) [reply cmake/FindParallelSTL.cmake:66]: Just in case we use the all caps version of the variable name.

**Review by umangyadav — APPROVED** [review]
